File: project_euler/Problem019/Problem019.go
package project_euler

import "fmt"

func Problem019() {
    start := 1900
    end := 2000
    days := []int{31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31}
    length := len(days)

    week := 1
    for _, day := range days {
        week += day
    }
    if 0 == start % 4 && (0 != start % 100 || 0 == start % 400) {
        week++
    }
    week %= 7

    result := 0
    for i := start + 1; i <= end; i++ {
        for j := 0; j < length; j++ {
            if 0 == week {
                result++
            }

            week += days[j]
            if 1 == j && (0 == i % 4 && (0 != i % 100 || 0 == i % 400)) {
                week++
            }
            week %= 7
        }
    }
    fmt.Println(result)
}

File: project_euler/Problem019/Problem019.py
class Problem019:
    def solution(self) -> None:
        start, end, days = 1900, 2000, [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31]
        length = len(days)

        week = 1
        for day in days:
            week += day
        if not start % 4 and (start % 100 or not start % 400):
            week += 1
        week %= 7

        result = 0
        for i in range(start + 1, end + 1):
            for j in range(length):
                if not week:
                    result += 1

                week += days[j]
                if 1 == j and (not start % 4 and (start % 100 or not start % 400)):
                    week += 1
                week %= 7
        print(result)
